### PR TITLE
[FLINK-31549][jdbc-driver] Add jdbc driver docs

### PR DESCRIPTION
**Pull request by FangYongs**
## What is the purpose of the change

Add jdbc driver docs

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no) no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no) no
  - The serializers: (yes / no / don't know) no
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know) no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know) no
  - The S3 file system connector: (yes / no / don't know) no

## Documentation

  - Does this pull request introduce a new feature? (yes / no) no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented) no

### CODE REVIEW

**Comment by FangYongs** [comment]
@libenchao I add docs of jdbc driver in this PR, please help to review it when you're free, thanks

File: docs/content.zh/docs/dev/table/jdbcDriver.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+### Use with Beeline
+
+Beeline is the command line tool for accessing [Apache Hive](https://hive.apache.org/), but it also supports general JDBC drivers. To install Hive and beeline, see [Hive documentation](https://cwiki.apache.org/confluence/display/Hive/GettingStarted#GettingStarted-RunningHiveServer2andBeeline.1).
+
+1. Download flink-jdbc-driver-bundle-{VERSION}.jar and add it to `$HIVE_HOME/lib`.
+2. Run beeline and connect to a Flink SQL gateway. As Flink SQL gateway currently ignores user names and passwords, just leave them empty.
+    ```
+    beeline> !connect jdbc:flink://localhost:8083
+    ```
+3. Execute any statement you want.
```
Comment (snuyanzin) [inline docs/content.zh/docs/dev/table/jdbcDriver.md:45]: There is a much lighter brother-twilling tool https://github.com/julianhyde/sqlline
they are both forks of the same initial tool however beeline has lots of hive dependencies while sqlline is agnostic

Do you mind to add docs for sqlline (also comes/used in Apache Calcite) ?

Moreover it has a number of features like syntax highlighting, autocompeteion, line numbers and so on
demo of some of them could be found at https://github.com/julianhyde/sqlline/wiki/Demos

Comment (FangYongs) [reply docs/content.zh/docs/dev/table/jdbcDriver.md:45]: Thanks @snuyanzin , sounds good to me and I have add an example for sqlline in doc

File: flink-table/flink-sql-jdbc-driver/src/main/java/org/apache/flink/table/jdbc/BaseConnection.java [hunk 2]
```diff
@@ -102,12 +101,6 @@ public int getTransactionIsolation() throws SQLException {
                 "FlinkConnection#getTransactionIsolation is not supported yet.");
     }
 
-    @Override
-    public SQLWarning getWarnings() throws SQLException {
```
Comment (libenchao) [inline flink-table/flink-sql-jdbc-driver/src/main/java/org/apache/flink/table/jdbc/BaseConnection.java:106]: If these changes are not relevant with this issue, I'd suggest to move these to another commit (if they are trivial, we can just add a hotfix commit; if they are major changes, we should create a Jira and link to it)

File: docs/content/docs/dev/table/jdbcDriver.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+# Flink JDBC Driver
+
+Flink JDBC Driver is a Java library for connecting and submitting SQL statements to [SQL Gateway]({{< ref "docs/dev/table/sql-gateway/overview" >}}) as the JDBC server.
+
+# Usage
+
+Before using Flink JDBC driver, you need to start a SQL Gateway as the JDBC server and binds it with your Flink cluster. We now assume that you have a gateway started and connected to a running Flink cluster.
+
+## Use with a JDBC Tool
```
[the remark this answers is not in the document]
Comment (FangYongs) [reply docs/content/docs/dev/table/jdbcDriver.md:35]: Thanks @libenchao Added

File: docs/content/docs/dev/table/jdbcDriver.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+### Use with Beeline
+
+Beeline is the command line tool for accessing [Apache Hive](https://hive.apache.org/), but it also supports general JDBC drivers. To install Hive and beeline, see [Hive documentation](https://cwiki.apache.org/confluence/display/Hive/GettingStarted#GettingStarted-RunningHiveServer2andBeeline.1).
+
+1. Download flink-jdbc-driver-bundle-{VERSION}.jar and add it to `$HIVE_HOME/lib`.
```
[the remark this answers is not in the document]
Comment (FangYongs) [reply docs/content/docs/dev/table/jdbcDriver.md:40]: Add dependencies section

File: docs/content/docs/dev/table/jdbcDriver.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+| 1 | Hi    |
+| 2 | Hello |
+| 1 | Hi    |
+| 2 | Hello |
++---+-------+
+4 rows selected (1.955 seconds)
+0: jdbc:flink://localhost:8083>
+```
+
+## Use with Java
```
[the remark this answers is not in the document]
Comment (FangYongs) [reply docs/content/docs/dev/table/jdbcDriver.md:129]: Add summary section

File: docs/content/docs/dev/table/jdbcDriver.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+**Sample.java**
+```java
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+public class Sample {
+	public static void main(String[] args) throws Exception {
+		try (Connection connection = DriverManager.getConnection("jdbc:flink://localhost:8083")) {
```
[the remark this answers is not in the document]
Comment (FangYongs) [reply docs/content/docs/dev/table/jdbcDriver.md:153]: Don't need it

File: docs/content/docs/dev/table/jdbcDriver.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
++----------------------------------+
+1 row selected (1.282 seconds)
+0: jdbc:flink://localhost:8083> SELECT * FROM T;
++---+-------+
+| a |   b   |
++---+-------+
+| 1 | Hi    |
+| 2 | Hello |
+| 1 | Hi    |
+| 2 | Hello |
```
[the remark this answers is not in the document]
Comment (FangYongs) [reply docs/content/docs/dev/table/jdbcDriver.md:123]: DONE

File: docs/content/docs/dev/table/jdbcDriver.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+```java
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+public class Sample {
+	public static void main(String[] args) throws Exception {
+		try (Connection connection = DriverManager.getConnection("jdbc:flink://localhost:8083")) {
+			try (Statement statement = connection.createStatement()) {
```
[the remark this answers is not in the document]
Comment (FangYongs) [reply docs/content/docs/dev/table/jdbcDriver.md:154]: Added `DataSource` example

File: flink-table/flink-sql-jdbc-driver/src/main/java/org/apache/flink/table/jdbc/BaseConnection.java (hunk 2 above)
Comment (FangYongs) [reply flink-table/flink-sql-jdbc-driver/src/main/java/org/apache/flink/table/jdbc/BaseConnection.java:106]: DONE

Comment (FangYongs) [reply flink-table/flink-sql-jdbc-driver/src/main/java/org/apache/flink/table/jdbc/BaseConnection.java]: I have move these commits to https://github.com/apache/flink/pull/22786

**Review by libenchao — APPROVED** [review]
+1 from me, @snuyanzin is there any more comments from your side?

**Review by snuyanzin — APPROVED** [review]
LGTM from my side

thanks for addressing comments and asking
